scripts/foundationdb/init-foundationdb.sh

```diff
@@ -0,0 +1,238 @@
+FDB_CLIENTS_MD5='f701c23c144cdee2a2bf68647f0e108e'
+log "Making sure ${FDB_CLIENTS_PKG} is in ${APT_CACHE}"
+cachepackage "${FDB_CLIENTS_PKG}" "${FDB_CLIENTS_MD5}"
+
+log "Installing ${FDB_CLIENTS_PKG}"
+dpkg --install ${APT_CACHE}/foundationdb-clients_6.1.8-1_amd64.deb
+
+
+### Installing FDB server package ###
+#-----------------------------------#
+FDB_SERVER_PKG='foundationdb-server_6.1.8-1_amd64.deb'
+FDB_SERVER_MD5='80a427be14a329d864a91c9ce464d73c'
+log "Making sure ${FDB_SERVER_PKG} is in ${APT_CACHE}"
+cachepackage "${FDB_SERVER_PKG}" "${FDB_SERVER_MD5}"
+
+log "Installing ${FDB_SERVER_PKG}"
+dpkg --install ${APT_CACHE}/foundationdb-server_6.1.8-1_amd64.deb
+
+
+### Getting rid of init.d management ###
+#--------------------------------------#
+log 'Making sure init.d does not manage FoundationDB'
+/etc/init.d/foundationdb stop
+update-rc.d foundationdb disable
+# rm /etc/init.d/foundationdb
+
+
+### Ensuring FDB directories are accessible ###
+#---------------------------------------------#
+log 'Making sure FDB directories are created and are owned by foundationdb user.'
+mkdir -pv /var/run/foundationdb
+chown foundationdb:foundationdb /var/run/foundationdb
+mkdir -pv /var/log/foundationdb
+chown -R foundationdb:foundationdb /var/log/foundationdb
+mkdir -pv "${DATA_DIR}"
+chown -R foundationdb:foundationdb "${DATA_DIR}"
+
+
+### Filling /etc/foundationdb/fdb.cluster ###
+#-------------------------------------------#
+CLUSTER_FILE=/etc/foundationdb/fdb.cluster
+log "Filling ${CLUSTER_FILE} file"
+if [ -f "${CLUSTER_FILE}" ]; then
+  cp ${CLUSTER_FILE} "${CLUSTER_FILE}.$(date +'%Y-%m-%d_%H-%M-%S')"
+fi
+echo "${CLUSTER_FILE_CONTENT}" > /etc/foundationdb/fdb.cluster
+
+
+### Filling /etc/foundationdb/foundationdb.conf ###
+#-------------------------------------------------#
+CONF_FILE=/etc/foundationdb/foundationdb.conf
+log "Filling ${CONF_FILE} file"
+if [ -f "${CONF_FILE}" ]; then
+  cp ${CONF_FILE} "${CONF_FILE}.$(date +'%Y-%m-%d_%H-%M-%S')"
+fi
+export HOST_TO_LISTEN_ON
+export DATA_DIR
+export FDB_SERVERS=$(
+  for server_id in $(seq 4500 $((4500 + SERVER_PROCESSES_NUM - 1))); do
+    echo "[fdbserver.${server_id}]"
+  done
+)
+envsubst '$HOST_TO_LISTEN_ON $DATA_DIR $FDB_SERVERS'\
+ < "${SCRIPT_DIR}/foundationdb.conf" > "${CONF_FILE}"
+
+
+### Defining systemd service ###
+#------------------------------#
+log 'Configuring systemd to manage FoundationDB'
+UNIT_FILE=/etc/systemd/system/foundationdb.service
+log "Filling ${UNIT_FILE} file"
+if [ -f "${UNIT_FILE}" ]; then
+  cp ${UNIT_FILE} "${UNIT_FILE}.$(date +'%Y-%m-%d_%H-%M-%S')"
+fi
+# Give FDB processes at most half of total memory (kill if greater).
+TOTAL_MEM_KB=$(awk '/MemTotal/ { print $2 }' /proc/meminfo)
+FDB_MEM_MAX=$(echo "$TOTAL_MEM_KB" | awk '{ printf "%d", $1 * 0.9 / 1024 }')
+DATA_DIRS=''
+for server_id in $(seq 4500 $((4500 + SERVER_PROCESSES_NUM - 1))); do
+  DATA_DIRS="${DATA_DIRS}${DATA_DIR}/${server_id} "
+done
+export MEMORY_MAX="${FDB_MEM_MAX}M"
```

Comment:
I think leaving the default configured memory limit (imposed by FDB itself) is fine for dev/test use cases.

We may want to configure the storage engine as `ssd` rather than the default so that available memory is less of a consideration.